packages/typescript/src/lib/Typescript.ts

```diff
@@ -190,17 +198,17 @@ export class FederatedTypesPlugin {
 
     try {
       // Try to resolve exposed component using index
-      const files = fs.readdirSync(path.join(rootDir, entry));
+      const files = this.getFiles(path.join(rootDir, entry));
 
-      filename = files.find((file) => file.split('.')[0] === 'index');
+      filename = files?.find((file) => file.split('.')[0] === 'index');
```

Comment:
Can you also handle the case where `filename` becomes **undefined**.

This will cause below line (`${entry}/${filename}`) be resolved to `path/to/undefined`.